### PR TITLE
change minor mistake of spelling

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: allencloud <allen.sun@daocloud.io>

fix typos in docs, test, logs and something others.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM 🐸
/cc @thaJeztah

File: docs/reference/run.md [hunk 1]
```diff
@@ -836,7 +836,7 @@ The container has unlimited memory which can cause the host to run out memory
 and require killing system processes to free memory. The `--oom-score-adj`
 parameter can be changed to select the priority of which containers will
 be killed when the system is out of memory, with negative scores making them
-less likely to be killed an positive more likely.
+less likely to be killed a positive more likely.
```
Comment (duglin) [inline docs/reference/run.md]: can you add a "," before the "a" just so its easier to parse - since you're in here anyway.

File: client/errors.go [hunk 2]
```diff
@@ -244,7 +244,7 @@ func (e secretNotFoundError) Error() string {
 	return fmt.Sprintf("Error: no such secret: %s", e.name)
 }
 
-// NoFound indicates that this error type is of NotFound
+// NotFound indicates that this error type is of NotFound
```
Comment (thaJeztah) [inline client/errors.go]: interesting, shouldn't golint check for this?

File: docs/reference/run.md (hunk 1 above)
Comment (thaJeztah) [inline docs/reference/run.md]: I think this should be

```Markdown
with negative scores making them
less likely to be killed, and positive scores more likely.
```

Actually, could you move this to a separate PR, so that this can be cherry-picked into the docs separately? (sorry, nitpicking). I know there's a rewrite coming up for the reference/run.md, in which case it may be good to be able to cherry pick this as well so that the cherry-pick goes in cleanly

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: client/errors.go (hunk 2 above)
Comment (aaronlehmann) [inline client/errors.go]: The type is not exported, so golint doesn't care about comment formatting.

File: api/server/router/swarm/backend.go [hunk 3]
```diff
@@ -7,7 +7,7 @@ import (
 	"golang.org/x/net/context"
 )
 
-// Backend abstracts an swarm commands manager.
+// Backend abstracts a swarm commands manager.
```
Comment (aaronlehmann) [inline api/server/router/swarm/backend.go]: Should this say "Backend abstracts a swarm manager"?

**Comment by allencloud** [comment]
Thanks @thaJeztah  Another PR for the doc correction has been submitted. See https://github.com/docker/docker/pull/29569

File: api/server/router/swarm/backend.go (hunk 3 above)
Comment (allencloud) [inline api/server/router/swarm/backend.go]: Updated. Thanks for feedback.

File: api/types/types.go [hunk 4]
```diff
@@ -154,8 +154,9 @@ type Version struct {
 	BuildTime     string `json:",omitempty"`
 }
 
-// Commit records a external tool actual commit id version along the
-// one expect by dockerd as set at build time
+// Commit records commits of an external tool, e.g containerd, runC.
+// ID represents the actual commit ID of external tool.
+// Expected represents the commit ID of external tool expected by dockerd as set at build time.
```
Comment (thaJeztah) [inline api/types/types.go]: Still not sure if it's clear with this description (what are "commits of an external tool"?). Just some quick suggestion, but not sure if it's any good;

```Markdown
Commit holds the Git-commit (SHA1), as reported in the version-string
of external tools, such as containerd, or runC.
```

Also; should `ID` and `Expected` be documented in this comment, or separately (as a comment _inside_ the struct?

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM
